gateway/src/cluster/impl.rs

```diff
@@ -135,7 +135,7 @@ impl Cluster {
     }
 
     pub(super) async fn new_with_config(mut config: Config) -> Result<Self, ClusterStartError> {
-        let [from, to, total] = match config.shard_scheme() {
+        let [from, to, step, total] = match config.shard_scheme() {
```

Comment:
This is getting a bit unruly, it might be good to break this match out into a function at this point, that way we can also add tests to ensure that `ShardScheme`s are translated into the proper `[from, to, step, total]` array